### PR TITLE
match to skate latest api and fix preact unmounting cleanup

### LINKED ISSUE #5: Make sure we notify preact in disconnectedCallback
So that `componentWillUnmount` lifecycle methods are properly called on any child preact components.

See skatejs/skatejs#1244

### PR DESCRIPTION
**Pull request by Hotell**
Closes #5

- applied preact hack

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -13,10 +13,16 @@ export default (Base = HTMLElement) =>
...
+      this._renderRoot = renderRoot
       this._preactDom = render(
         renderCallback(),
         renderRoot,
         this._preactDom || renderRoot.children[0]
       );
     }
+    disconnectedCallback() {
+      super.disconnectedCallback && super.disconnectedCallback();
+      this._preactDom = render(<div/>,this._renderRoot,this._preactDom);
```
Comment (Hotell) [inline src/index.js]: original is correctly replaced with empty div, although Preact won't call `componentWillUnmount` on mounted wrapped component ...

Comment (Hotell) [reply src/index.js]: Also it's not a shadowDOM thing. Following wont trigger willUnmount as well ( pure Preact )

```js
import { h, Component, render } from 'preact'
class App extends Component {
  render(){
    return <div>original</div>
  }
  componentDidMount(){
    console.log('hi')
  }
  componentWillUnmount(){
    console.log('bye!')
  }
}
const mountTo = document.getElementById("root")
let root;

  root = render(<App />, mountTo);
  console.log(root)
  setTimeout(()=>{
   // element is replaced bud willUnmount won't be called ... Preact BUG ?
    render(<div>replaced</div>, mountTo, root)
  },1400)
```

Comment (treshugart) [reply src/index.js]: Yeah, I don't have any experience here. Might be worth pinging @developit about this? It looks like unmounting at a node is in compat, but I don't want to place a dependency on that https://github.com/developit/preact/issues/53.

**Comment by Hotell** [comment]
Re-review pls @treshugart

**Comment by treshugart** [comment]
Cool, thanks @Hotell!

**Pull request merged**